config/config_test.go

```diff
@@ -873,153 +782,266 @@ func TestCompressionValidateAndLog(t *testing.T) {
 	}
 
 	testCases := []struct {
-		description     string
-		compConf        *Compression
-		expectFatal     bool
-		expectedLogInfo []logComponents
+		description      string
+		inCompressionCfg *Compression
+		expectedLogInfo  []logComponents
+	}{
+		{
+			description:      "Blank compression type, expect fatal level log entry",
+			inCompressionCfg: &Compression{Type: CompressionType("")},
+			expectedLogInfo: []logComponents{
+				{msg: "invalid config.compression.type: . It must be \"none\" or \"snappy\"", lvl: logrus.FatalLevel},
+			},
+		},
+		{
+			description:      "Valid compression type 'none', expect info level log entry",
+			inCompressionCfg: &Compression{Type: CompressionNone},
+			expectedLogInfo: []logComponents{
+				{msg: "config.compression.type: none", lvl: logrus.InfoLevel},
+			},
+		},
+		{
+			description:      "Valid compression type 'snappy', expect info level log entry",
+			inCompressionCfg: &Compression{Type: CompressionSnappy},
+			expectedLogInfo: []logComponents{
+				{msg: "config.compression.type: snappy", lvl: logrus.InfoLevel},
+			},
+		},
+		{
+			description:      "Unsupported compression, expect fatal level log entry",
+			inCompressionCfg: &Compression{Type: CompressionType("UnknownCompressionType")},
+			expectedLogInfo: []logComponents{
+				{msg: "invalid config.compression.type: UnknownCompressionType. It must be \"none\" or \"snappy\"", lvl: logrus.FatalLevel},
+			},
+		},
+	}
+
+	//substitute logger exit function so execution doesn't get interrupted
+	defer func() { logrus.StandardLogger().ExitFunc = nil }()
+	logrus.StandardLogger().ExitFunc = func(int) {}
+
+	for _, tc := range testCases {
+		// Run test
+		tc.inCompressionCfg.validateAndLog()
+
+		// Assert logrus expected entries
+		if !assert.Len(t, hook.Entries, len(tc.expectedLogInfo), "Incorrect number of entries were logged to logrus. Expected: %d. Actual: %d", len(tc.expectedLogInfo), len(hook.Entries)) {
+			return
```

Comment:
Instead of a return, how about something like this:

```
if assert.Len(t, hook.Entries, len(tc.expectedLogInfo), tc.description) {
	for i := 0; i < len(tc.expectedLogInfo); i++ {
		assert.Equal(t, tc.expectedLogInfo[i].msg, hook.Entries[i].Message, tc.description+":message")
		assert.Equal(t, tc.expectedLogInfo[i].lvl, hook.Entries[i].Level, tc.description+":level")
	}
}
```